Server/mods/deathmatch/logic/CCustomData.cpp

```diff
@@ -75,23 +65,30 @@ void CCustomData::UpdateSynced(const char* szName, const CLuaArgument& Variable,
         }
     }
     else
-    {
         DeleteSynced(szName);
-    }
 }
 
-void CCustomData::Set(const char* szName, const CLuaArgument& Variable, bool bSynchronized)
+// Returns false in case we try to set the same value for the custom data and we didn't change bSync, otherwise true
+bool CCustomData::Set(const char* szName, const CLuaArgument& Variable, bool bSynchronized, CLuaArgument* pOldVariable)
 {
     assert(szName);
 
     // Grab the item with the given name
-    SCustomData* pData = Get(szName);
+    SCustomData* pData = MapFind(m_Data, szName);

```

Comment:
Get should still be used here. That way if the underlying data structure changes, fewer references to m_Data need to be changed